app/controllers/incomeSources/add/BusinessAccountingMethodController.scala

```diff
@@ -0,0 +1,171 @@
+                                                   val retrieveNino: NinoPredicate,
+                                                   val view: BusinessAccountingMethod,
+                                                   val customNotFoundErrorView: CustomNotFoundError)
+                                                  (implicit val appConfig: FrontendAppConfig,
+                                                   mcc: MessagesControllerComponents,
+                                                   val ec: ExecutionContext,
+                                                   val itvcErrorHandler: ItvcErrorHandler,
+                                                   val itvcErrorHandlerAgent: AgentItvcErrorHandler)
+  extends ClientConfirmedController with FeatureSwitching with I18nSupport {
+
+  def handleRequest(isAgent: Boolean)
+                   (implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext, messages: Messages): Future[Result] = {
+
+    val incomeSourcesEnabled: Boolean = isEnabled(IncomeSources)
+    val backUrl: String = if (isAgent) controllers.incomeSources.add.routes.CheckBusinessDetailsController.showAgent().url else
+      controllers.incomeSources.add.routes.CheckBusinessDetailsController.show().url
+    val postAction: Call = if (isAgent) controllers.incomeSources.add.routes.BusinessAccountingMethodController.submitAgent() else
+      controllers.incomeSources.add.routes.BusinessAccountingMethodController.submit()
+
+    val errorHandler: ShowInternalServerError = if (isAgent) itvcErrorHandlerAgent else itvcErrorHandler
+
+    if (incomeSourcesEnabled) {
+      val userBusinesses: List[BusinessDetailsModel] = user.incomeSources.businesses
+      if (shouldAutomaticallyRedirect(userBusinesses)) {
+        if (isAgent) {
+          Future.successful(Redirect(controllers.incomeSources.add.routes.CheckBusinessDetailsController.showAgent())
+            .addingToSession(addBusinessAccountingMethod -> "cash"))
+        } else {
+          Future.successful(Redirect(controllers.incomeSources.add.routes.CheckBusinessDetailsController.show())
+            .addingToSession(addBusinessAccountingMethod -> "cash"))
+        }
+      } else {
+        Future.successful(Ok(view(
+        form = BusinessAccountingMethodForm.form,
+        postAction = postAction,
+        isAgent = isAgent,
+        backUrl = backUrl,
+        btaNavPartial = user.btaNavPartial
+        )(user, messages)))
+      }
+    } else {
+      Future.successful(Ok(customNotFoundErrorView()(user, messages)))
+    } recover {
+      case ex: Exception =>
+        Logger("application").error(s"${if (isAgent) "[Agent]"}" +
+          s"Error getting BusinessEndDate page: ${ex.getMessage}")
+        errorHandler.showInternalServerError()
+    }
+  }
+
+  def shouldAutomaticallyRedirect(businesses: List[BusinessDetailsModel]): Boolean = {
+    val activeSoleTraderBusinesses: Boolean = !businesses.forall(_.isCeased)
```

Comment:
this variable definition is redundant, used only once

Reply:
I think it's good for readability because this function is just the condition inside an if statement to decide whether to automatically redirect or not